### PR TITLE
Patient attachments app improvement

### PR DESCRIPTION
**Pull request by vasharma05**
## Requirements

- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [x] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide).
- [x] My work includes tests or is validated by existing tests.

## Summary
Improvement to the functionality and UI/ UX for patient attachments.

- [x] Camera capturing improved
- [x] Uploading files and reviewing the files to change their name and description completed.

## Screenshots

None yet

### CODE REVIEW

**Comment by ZacButko** [comment]
@vasharma05 looks reasonable so far 👍

File: packages/esm-patient-attachments-app/src/attachments/attachments-grid-overview.scss [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+.rightPanel {
+  background-color: #393939;
+  padding: $spacing-05;
+  color: $ui-01;
+}
+
+.closePreviewButton svg path {
+  fill: $ui-01 !important;
```
Comment (ZacButko) [inline packages/esm-patient-attachments-app/src/attachments/attachments-grid-overview.scss:44]: no important

File: packages/esm-patient-attachments-app/src/attachments/attachments-grid-overview.scss [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+.closePreviewButton:hover svg path {
+  fill: $ui-05 !important;
```
Comment (ZacButko) [inline packages/esm-patient-attachments-app/src/attachments/attachments-grid-overview.scss:48]: no important

File: packages/esm-patient-attachments-app/src/attachments/camera-upload.scss [hunk 3]
```diff
@@ -10,30 +11,67 @@
...
   float: right;
   color: #0f62fe;
   cursor: pointer;
   margin: 1rem 0;
+  padding: 1rem;
 
   &:hover {
     color: #000;
     text-decoration: none;
+    background-color: #e5e5e5;
```
Comment (ZacButko) [inline packages/esm-patient-attachments-app/src/attachments/camera-upload.scss:24]: no magic colors

File: packages/esm-patient-attachments-app/src/attachments/camera-upload.scss [hunk 4]
```diff
@@ -10,30 +11,67 @@
...
+  overflow-y: auto;
+  border: 1px dashed #8d8d8d;
+}
+
+.buttonSet {
+  width: 100%;
+  display: grid;
+  grid-template-columns: 1fr 1fr;
+  button{
+    max-width: unset !important;
```
Comment (ZacButko) [inline packages/esm-patient-attachments-app/src/attachments/camera-upload.scss:75]: no important

File: packages/esm-patient-attachments-app/src/attachments/camera-upload.scss [hunk 5]
```diff
@@ -10,30 +11,67 @@
...
+
+.uploadingFilesSection {
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+  margin: $spacing-05;
+  height: 25rem;
+  overflow-y: auto;
+  border: 1px dashed #8d8d8d;
```
Comment (ZacButko) [inline packages/esm-patient-attachments-app/src/attachments/camera-upload.scss:67]: magic color

File: packages/esm-patient-attachments-app/src/attachments/camera-upload.scss [hunk 6]
```diff
@@ -10,30 +11,67 @@
...
   &:hover {
     color: #000;
     text-decoration: none;
+    background-color: #e5e5e5;
   }
 }
 
 .uploadFile {
-  display: none;
+  color: #fff;
```
Comment (ZacButko) [inline packages/esm-patient-attachments-app/src/attachments/camera-upload.scss:29]: magic color

File: packages/esm-patient-attachments-app/src/attachments/image-preview.scss [hunk 7]
```diff
@@ -1,18 +1,33 @@
...
 
-.buttonSetOverrides button {
-  max-width: 20rem !important;
+.buttonSetOverrides {
+  margin: $spacing-05 auto 0;
+  display: grid;
+  grid-template-columns: 1fr 1fr;
+
+  button {
+    max-width: unset !important;
```
Comment (ZacButko) [inline packages/esm-patient-attachments-app/src/attachments/image-preview.scss:29]: no important

**Review by ZacButko — CHANGES_REQUESTED** [review]
A bunch of small css cleanup. Otherwise looks good

File: packages/esm-patient-attachments-app/src/attachments/attachments-grid-overview.scss [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+.galleryContainer {
+  display: grid;
+  grid-template-columns: 1fr 1fr 1fr 1fr;
+  align-items: center;
+  grid-gap: $spacing-05;
+  padding: 0 $spacing-05 $spacing-05 $spacing-05;
+}
+
+.attachmentThumbnailSkeleton {
+  width: 100% !important;
```
[the remark this answers is not in the document]
Comment (vasharma05) [reply packages/esm-patient-attachments-app/src/attachments/attachments-grid-overview.scss:12]: Thank you for pointing this out.

File: packages/esm-patient-attachments-app/src/attachments/attachments-grid-overview.scss [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+.leftPanel {
+  display: grid;
+  grid-template-columns: auto 1fr auto;
+  align-items: start;
+}
+
+.rightPanel {
+  background-color: #393939;
```
[the remark this answers is not in the document]
Comment (vasharma05) [reply packages/esm-patient-attachments-app/src/attachments/attachments-grid-overview.scss:38]: Replaced with `$carbon--gray-80`. Thank you for the help @ZacButko!

File: packages/esm-patient-attachments-app/src/attachments/attachments-grid-overview.component.tsx [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+        <SkeletonPlaceholder className={styles.attachmentThumbnailSkeleton} />
+        <SkeletonPlaceholder className={styles.attachmentThumbnailSkeleton} />
+        <SkeletonPlaceholder className={styles.attachmentThumbnailSkeleton} />
+      </div>
+    );
+  }
+
+  return (
+    <div className={styles.galleryContainer}>
+      {attachments.map((attachment, indx) => {
```
Comment (ZacButko) [inline packages/esm-patient-attachments-app/src/attachments/attachments-grid-overview.component.tsx:32]: 5 letters for `index` was too expensive? 😂

**Review by ZacButko — APPROVED** [review]

File: packages/esm-patient-attachments-app/src/attachments/attachments-grid-overview.component.tsx (hunk 10 above)
Comment (ibacher) [reply packages/esm-patient-attachments-app/src/attachments/attachments-grid-overview.component.tsx:32]: Gotta save those byts

Comment (vasharma05) [reply packages/esm-patient-attachments-app/src/attachments/attachments-grid-overview.component.tsx:32]: 😆

**Pull request merged**